src/equations/canvas.cljs

```diff
@@ -68,17 +68,44 @@
 
     ctx))
 
+(defn add-point! [graph coords]
+  (let [ctx (:context graph)
+        x-coord (first (goog.object/get coords "tail"))
+        y-coord (second (goog.object/get coords  "tail"))]
+
+    (set! (.-fillStyle ctx) "red")
+    (.beginPath ctx)
+    (.arc ctx x-coord y-coord (/ 3 (:scale-x graph)) 0 (* js/Math.PI 2) true)
+    (.fill ctx)
+    ctx))
+
 ;; events
 
 (rf/reg-event-db
   :initialize
   (fn [_ _]
     {:equations []
-     :animate false}))
+     :animate false
+     :coords []}))
 
 (defn re-trigger-timer []
   (r/next-tick (fn [] (rf/dispatch [:timer]))))
 
+(defn canvas-click [event graph]
+  (let [rect (.getBoundingClientRect (:canvas graph))
+        x-coord (/ (-
+                     (goog.object/get (goog.object/get event "nativeEvent") "clientX")
+                     (goog.object/get rect "left")
+                     (:center-x graph))
```

Comment:
Introducing a temporary variable for the pixel-scale `x` value here might make this clearer to me.